<launch>

    <!-- Camera description  -->
    <include file="$(find rgbd_camera_util)/launch/asus.launch"/>
    <include file="$(find rgbd_camera_util)/launch/display.launch"/>

    <!-- Static transform between fixed frame (map) and sensor base -->
    <node pkg="tf" type="static_transform_publisher" name="world_map_tf" args="0 0 0 0 0 0 map world 30" />

		<!-- Run LSD-SLAM for monocular cameras (NOT WORKING IN ROSLAUNCH)
		<node pkg="lsd_slam_core" type="live_slam" name="live_slam">
    	<remap from="/image" to="/camera/rgb/image_rect_color"/>
    	<remap from="/camera_info" to="/camera/rgb/camera_info"/>
		</node> 

		IN COMMAND LINE
		rosrun lsd_slam_core live_slam /image:=/camera/rgb/image_rect_color /camera_info:=/camera/rgb/camera_info

     -->

		<!-- Generate colored PCL (UNNECESSARY)-->
    <node pkg="nodelet" type="nodelet" args="manager" name="nodelet_manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="my_cloudify_with_color"
          args="load depth_image_proc/point_cloud_xyzrgb nodelet_manager no-bond">
      <remap from="rgb/camera_info" to="/camera/rgb/camera_info"/>
      <remap from="rgb/image_rect_color" to="/camera/rgb/image_rect_color"/> 
      <remap from="depth_registered/image_rect" to="/camera/depth/image_rect_raw"/>
      <remap from="depth_registered/points" to="/original_colored_pcl"/>
    </node> 

    <!-- Run octomap -->
		<node pkg="octomap_server" type="octomap_server_node" name="octomap_server" output="screen">
        <param name="resolution" value="0.01" />     
        <param name="frame_id" type="string" value="/world" />
        <param name="base_frame_id" type="string" value="/lsd_slam/pose" />
        <param name="sensor_model/max_range" value="5.0" />
        <remap from="cloud_in" to="/camera/depth/points" />
    </node> 

		<!-- Run Rviz and load the default config to see the state of the move_group node -->
  	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find semantic_mapping)/rviz/lsd_slam.rviz"/> 

</launch>
